analyzers/src/SonarAnalyzer.Common/Rules/TooManyParametersBase.cs

```diff
@@ -50,7 +51,10 @@ public abstract class TooManyParametersBase<TSyntaxKind, TParameterListSyntax> :
                 {
                     var parametersCount = CountParameters((TParameterListSyntax)c.Node);
                     var baseCount = BaseParameterCount(c.Node.Parent);
-                    if (parametersCount - baseCount > Maximum && c.Node.Parent != null && CanBeChanged(c.Node.Parent, c.SemanticModel))
+                    if (parametersCount - baseCount > Maximum
+                        && c.Node.Parent is { } parent
+                        && !IsExtern(parent)
+                        && CanBeChanged(parent, c.SemanticModel))
```

Comment:
CanBeChanged retrieves the method symbol. The method symbol also has some indications of "extern":
* [IMethodSymbol.MethodImplementationFlags](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.imethodsymbol.methodimplementationflags?view=roslyn-dotnet-4.6.0)
* [IMethodSymbol.MethodKind](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.imethodsymbol.methodkind?view=roslyn-dotnet-4.6.0) (MethodKind.DeclareMethod)

I don't know if either of these properties help to identify extern methods. What do we want to do about VB support? create an issue? If so, we could add this information there.

Reply:
If you agree, given that VB is no longer a priority, I would not even create the issue and wait for the community to raise concerns.